src/treelearner/monotone_constraints.hpp

```diff
@@ -8,31 +8,55 @@
 
 namespace LightGBM {
 
-struct LeafConstraints {
-  double min_constraint;
-  double max_constraint;
+struct ConstraintEntry {
+  double min = -std::numeric_limits<double>::max();
+  double max = std::numeric_limits<double>::max();
 
-  static void SetChildrenConstraintsFastMethod(
-    std::vector<LeafConstraints> &constraints_per_leaf, int *right_leaf,
-    int *left_leaf, int8_t monotone_type, double right_output,
-    double left_output, bool is_numerical_split);
+  ConstraintEntry(){};
 
-  LeafConstraints() {
-    Reset();
+  void Reset() {
+    min = -std::numeric_limits<double>::max();
+    max = std::numeric_limits<double>::max();
   }
 
+  void UpdateMin(double new_min) { min = std::min(new_min, min); }
```

Comment:
This should be: 
void UpdateMin(double new_min) { min = std::max(new_min, min); }
void UpdateMax(double new_max) { max = std::min(new_max, max); }

because when there are 2 min constraints, the relevant one is the greatest one, and when there are 2 maximum constraints, the relevant one is the smallest one.